// spotify-sdk/lib/index.js
'use strict';

Object.defineProperty(exports, '__esModule', {
	value: true
});

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { 'default': obj }; }

var _Client = require('./Client');

var _Client2 = _interopRequireDefault(_Client);

var _handlersArtistHandler = require('./handlers/ArtistHandler');

var _handlersArtistHandler2 = _interopRequireDefault(_handlersArtistHandler);

var _handlersAlbumHandler = require('./handlers/AlbumHandler');

var _handlersAlbumHandler2 = _interopRequireDefault(_handlersAlbumHandler);

var _handlersTrackHandler = require('./handlers/TrackHandler');

var _handlersTrackHandler2 = _interopRequireDefault(_handlersTrackHandler);

var _handlersPlaylistHandler = require('./handlers/PlaylistHandler');

var _handlersPlaylistHandler2 = _interopRequireDefault(_handlersPlaylistHandler);

var _handlersUserHandler = require('./handlers/UserHandler');

var _handlersUserHandler2 = _interopRequireDefault(_handlersUserHandler);

var _handlersCollectionHandler = require('./handlers/CollectionHandler');

var _handlersCollectionHandler2 = _interopRequireDefault(_handlersCollectionHandler);

var _entitiesArtistEntity = require('./entities/ArtistEntity');

var _entitiesArtistEntity2 = _interopRequireDefault(_entitiesArtistEntity);

var _entitiesAlbumEntity = require('./entities/AlbumEntity');

var _entitiesAlbumEntity2 = _interopRequireDefault(_entitiesAlbumEntity);

var _entitiesTrackEntity = require('./entities/TrackEntity');

var _entitiesTrackEntity2 = _interopRequireDefault(_entitiesTrackEntity);

var _entitiesPlaylistEntity = require('./entities/PlaylistEntity');

var _entitiesPlaylistEntity2 = _interopRequireDefault(_entitiesPlaylistEntity);

var _entitiesUserEntity = require('./entities/UserEntity');

var _entitiesUserEntity2 = _interopRequireDefault(_entitiesUserEntity);

var _entitiesCollectionEntity = require('./entities/CollectionEntity');

var _entitiesCollectionEntity2 = _interopRequireDefault(_entitiesCollectionEntity);

var _helpersArtist = require('./helpers/Artist');

var _helpersArtist2 = _interopRequireDefault(_helpersArtist);

var _helpersAlbum = require('./helpers/Album');

var _helpersAlbum2 = _interopRequireDefault(_helpersAlbum);

var _helpersTrack = require('./helpers/Track');

var _helpersTrack2 = _interopRequireDefault(_helpersTrack);

var _helpersPlaylist = require('./helpers/Playlist');

var _helpersPlaylist2 = _interopRequireDefault(_helpersPlaylist);

var _helpersUser = require('./helpers/User');

var _helpersUser2 = _interopRequireDefault(_helpersUser);

var _helpersCollection = require('./helpers/Collection');

var _helpersCollection2 = _interopRequireDefault(_helpersCollection);

exports['default'] = {
	Client: _Client2['default'],
	ArtistHandler: _handlersArtistHandler2['default'],
	AlbumHandler: _handlersAlbumHandler2['default'],
	TrackHandler: _handlersTrackHandler2['default'],
	PlaylistHandler: _handlersPlaylistHandler2['default'],
	UserHandler: _handlersUserHandler2['default'],
	CollectionHandler: _handlersCollectionHandler2['default'],
	ArtistEntity: _entitiesArtistEntity2['default'],
	AlbumEntity: _entitiesAlbumEntity2['default'],
	TrackEntity: _entitiesTrackEntity2['default'],
	PlaylistEntity: _entitiesPlaylistEntity2['default'],
	UserEntity: _entitiesUserEntity2['default'],
	CollectionEntity: _entitiesCollectionEntity2['default'],
	Artist: _helpersArtist2['default'],
	Album: _helpersAlbum2['default'],
	Track: _helpersTrack2['default'],
	Playlist: _helpersPlaylist2['default'],
	User: _helpersUser2['default'],
	Collection: _helpersCollection2['default']
};
module.exports = exports['default'];
