// spotify-sdk/lib/handlers/PlaylistHandler.js
/*
 * PlaylistHandler
 * Methods for retrieving information about playlists and for managing playlists.
 *
 * Doc: https://developer.spotify.com/web-api/playlist-endpoints/
 */
'use strict';

Object.defineProperty(exports, '__esModule', {
    value: true
});

var _createClass = (function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ('value' in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; })();

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { 'default': obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError('Cannot call a class as a function'); } }

var _helpersPlaylist = require('../helpers/Playlist');

var _helpersPlaylist2 = _interopRequireDefault(_helpersPlaylist);

var _Client = require('../Client');

var _Client2 = _interopRequireDefault(_Client);

var PlaylistHandler = (function () {
    function PlaylistHandler() {
        _classCallCheck(this, PlaylistHandler);
    }

    _createClass(PlaylistHandler, [{
        key: 'search',

        /*
         * Get a collection of playlists that match a keyword string.
         * Doc: https://developer.spotify.com/web-api/search-item/
         *
         * @public
         * @param {string} name Name of the playlist.
         * @param {object} query Optional query parameters.
         * @return {Collection} playlistCollection
         */
        value: function search(name, query) {
            return _Client2['default'].instance.request('/search?type=playlist&q=' + encodeURIComponent(name), 'GET', query);
        }

        /*
         * Get a list of Spotify featured playlists.
         * Doc: https://developer.spotify.com/web-api/get-list-featured-playlists/
         *
         * @public
         * @required {OAuth}
         * @param {object} query Optional query parameters.
         * @return {Collection} playlistCollection
         */
    }, {
        key: 'featuredPlaylists',
        value: function featuredPlaylists(query) {
            return _Client2['default'].instance.request('/browse/featured-playlists', 'GET', query);
        }

        /*
         * Get a list of Spotify playlists tagged with a particular category.
         * Doc: https://developer.spotify.com/web-api/get-categorys-playlists/
         *
         * @public
         * @required {OAuth}
         * @param {int} id Category id
         * @param {object} query Optional query parameters.
         * @return {Collection} playlistCollection
         */
    }, {
        key: 'categoriesPlaylists',
        value: function categoriesPlaylists(id, query) {
            return _Client2['default'].instance.request('/browse/categories/' + id + '/playlists', 'GET', query);
        }

        /*
         * Add one or more tracks to a user’s playlist.
         * Doc: https://developer.spotify.com/web-api/add-tracks-to-playlist/
         *
         * @public 
         * @required {OAuth}
         * @param {array} tracks List of uri tracks
         * @param {string} userId User id
         * @param {string} playlistId Playlist id
         * @param {object} query Optional query parameters.
         * @return {} 
         */
    }, {
        key: 'addTracks',
        value: function addTracks(tracks, userId, playlistId, query) {
            return _Client2['default'].instance.request('/users/' + userId + '/playlists/' + playlistId + '/tracks', 'POST', { uris: tracks });
        }

        /*
         * Remove one or more tracks from a user’s playlist.
         * Doc: https://developer.spotify.com/web-api/remove-tracks-playlist/
         *
         * @public 
         * @required {OAuth}
         * @param {array} tracks List of uri tracks
         * @param {string} userId User id
         * @param {string} playlistId Playlist id
         * @param {object} query Optional query parameters.
         * @return {} 
         */
    }, {
        key: 'removeTracks',
        value: function removeTracks(tracks, userId, playlistId, query) {
            return _Client2['default'].instance.request('/users/' + userId + '/playlists/' + playlistId + '/tracks', 'DELETE', { uris: tracks });
        }

        /*
         * Add a user as a follower of a playlist.
         * Doc: https://developer.spotify.com/web-api/follow-playlist/
         *
         * @public 
         * @required {OAuth}
         * @param {string} userId User id
         * @param {string} playlistId Playlist id
         * @param {object} query Optional query parameters.
         * @return {} 
         */
    }, {
        key: 'follow',
        value: function follow(userId, playlistId, query) {
            return _Client2['default'].instance.request('/users/' + userId + '/playlists/' + playlistId + '/followers', 'PUT');
        }

        /*
         * Remove a user as a follower of a playlist.
         * Doc: https://developer.spotify.com/web-api/unfollow-playlist/
         *
         * @public 
         * @required {OAuth}
         * @param {string} userId User id
         * @param {string} playlistId Playlist id
         * @param {object} query Optional query parameters.
         * @return {} 
         */
    }, {
        key: 'unfollow',
        value: function unfollow(userId, playlistId, query) {
            return _Client2['default'].instance.request('/users/' + userId + '/playlists/' + playlistId + '/followers', 'DELETE');
        }

        /*
         * Create a playlist for a Spotify user.
         * Doc: https://developer.spotify.com/web-api/create-playlist/
         *
         * @public 
         * @required {OAuth}
         * @param {string} userId User id
         * @param {string} name Name of the future playlist
         * @param {bool} public public or private
         * @param {object} query Optional query parameters.
         * @return {} 
         */
    }, {
        key: 'create',
        value: function create(userId, name, is_public) {
            return _Client2['default'].instance.request('/users/' + userId + '/playlists', 'POST', { name: name, 'public': is_public });
        }

        /*
         * Change a playlist’s name and public/private state.
         * Doc: https://developer.spotify.com/web-api/change-playlist-details/
         *
         * @public 
         * @required {OAuth}
         * @param {string} userId User id
         * @param {string} playlistId Playlist id
         * @param {string} name Name of the future playlist
         * @param {bool} public public or private
         * @param {object} query Optional query parameters.
         * @return {} 
         */
    }, {
        key: 'edit',
        value: function edit(userId, playlistId, name, is_public) {
            return _Client2['default'].instance.request('/users/' + userId + '/playlists', 'PUT', { name: name, 'public': is_public });
        }

        /*
         * @public 
         * @param {object} item Object to convert in entity
         * @return {Playlist}
         */
    }, {
        key: 'convert',
        value: function convert(item) {
            return new _helpersPlaylist2['default'](item);
        }
    }]);

    return PlaylistHandler;
})();

exports['default'] = PlaylistHandler;
module.exports = exports['default'];
